### PR TITLE
#169257397 User can login via email and password

### PR DESCRIPTION
**Pull request by RaymondMwaura**
#### What does this PR do?
This PR creates a login functionality for users to get access to the application by supplying their email and password

#### Description of Task to be completed?
- [x] Setup/refactor user model, route and controller
- [x] Setup jwtToken and Bcrypt
- [x] Setup seeder file to populate test user record in the database
- [x] Successfully link route with validation package

#### How should this be manually tested?
Option 1
- Clone this repo and navigate to the project folder using Terminal
- Install dependencies using the command `npm install`
- Run the command `npm test` on Terminal

Option 2
- Run the command `npx sequelize-cli db:seed:all` to add a test user to your database
- Using Postman, create a POST request to the URL: (http://localhost:3000/api/v1/auth/signin) with the following credentials { email: 'john@barefoot.com',  password: '12345678' }
- You should receive a success response, change the credentials to get an error message

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#169257397](https://www.pivotaltracker.com/story/show/169257397)

#### Screenshots (if appropriate)
successfulLogin

#### Questions:
N/A

### CODE REVIEW

File: src/routes/api/users.js [hunk 1]
```diff
@@ -1,23 +1,59 @@
...
+ *                     type: string
+ *                   token:
+ *                     type: string
+ *     responses:
+ *       200:
+ *         description: success
+ */
+
+user.post('/auth/signup', validation, checkForEmail, users.createUser);
+user.post('/auth/signin', validation, users.findUser);
```
Comment (b0nbon1) [inline src/routes/api/users.js:101]: Great work but Add swagger documentation for your endpoint, please

File: src/tests/users.test.js [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    }
+  });
+
+  it('should not login unregistered user', async () => {
+    try {
+      const res = await request(app)
+        .post(`${prefix}/auth/signin`)
+        .send({
+          email: 'unregistered@unregistered.com',
+          password: 'unregistered',
```
Comment (b0nbon1) [inline src/tests/users.test.js]: I think i would be better if you separated your tests from the signup ones for better unit testing

**Review by b0nbon1 — CHANGES_REQUESTED** [review]

File: src/tests/users.test.js (hunk 2 above)
Comment (RaymondMwaura) [reply src/tests/users.test.js]: O.k. Let me work on this.

Comment (b0nbon1) [reply src/tests/users.test.js]: Also resolve the conflicts from develop

Comment (RaymondMwaura) [reply src/tests/users.test.js]: I have made the changes.

**Review by nshaphan — DISMISSED** [review]
great work @RaymondMwaura

**Review by b0nbon1 — DISMISSED** [review]
I have tested and also gone through the code. Wonderful job, I think it is ready to be merged

**Review by timolinn — COMMENTED** [review]
Test failed with `ERROR: Unable to find migration: 20191104140247-create-user.js`

**Comment by RaymondMwaura** [comment]
> Test failed with `ERROR: Unable to find migration: 20191104140247-create-user.js`

You'll need a '.env' file with the variables below (the specific parameters, including the database names, should match your setup). You'll also need to manually drop the 'SequelizeMeta' table in your PostgreSQL instance that probably contains old migrations that are preventing the proper functioning of the current migration scripts. Then simply run `npm install` and `npm test`. You may have to do this for each PR you test.

DEV_DATABASE_URL=postgres://timothy:yourpassword@127.0.0.1:5432/yourdevdb
TEST_DATABASE_URL=postgres://timothy:yourpassword@127.0.0.1:5432/yourtestdb
SECRET=your_secret
SENDGRID_API_KEY=your_sendgrid_key

**Comment by timolinn** [comment]
> Test failed with `ERROR: Unable to find migration: 20191104140247-create-user.js`

Works now.

File: src/tests/userSignIn.test.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+import { use, request, expect } from 'chai';
+import app from '../app';
+import db from '../models';
+
+use(chaiHttp);
+
+const prefix = '/api/v1';
+describe('/auth/signin', () => {
+  before(async () => {
+    try {
```
Comment (timolinn) [inline src/tests/userSignIn.test.js]: Using a `try - catch` block causes your tests to never fail.

**Review by timolinn — CHANGES_REQUESTED** [review]
Remove try catch blocks from tests

**Review by timolinn — APPROVED** [review]

**Review by b0nbon1 — APPROVED** [review]
nice work.

**Pull request merged**